### PR TITLE
chore: fix list widget data parsing

### LINKED ISSUE #9872: [Bug]-[8000]:JSON.parse fails in List Widget, needs escaping special chars like single quotes and \r
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Screenshot 2021-12-20 at 3 13 28 PM

Screenshot 2021-12-20 at 3 14 25 PM

```js
originalBinding - "((currentItem) => { navigateTo(currentItem.task, {})})(JSON.parse('{"step":"#1","task":"Drop a t'able","status":"✅","action":"'"}'))"
```

Whenever data has a single quote in it, `JSON.parse`,
root cause 
```js
// In ListWidget/widget/index.tsx line no. 461
  `{{((currentItem) => { ${next}})(JSON.parse('${escapedStringifiedListItem}'))}}`
  
  // here we have single quotes
```

### Solution

1. Either change the way listWidget access its currentItem or 
2. Escape the single quotes and after JSON parsing unescape them.

### Steps To Reproduce

1. Add list widget to canvas and add a button to the item.
2. List widget options = 
```
[
  {
    "step": "#1",
    "task": "Drop a t'able",
    "status": "✅",
    "action": "'"
  },
  {
    "step": "#2",
    "task": "Create a query fetch_users with the Mock DB",
    "status": "--",
    "action": ""
  },
  {
    "step": "#3",
    "task": "Bind the query using => fetch_users.data",
    "status": "--",
    "action": ""
  }
]
```
3. Add onClick of button `{{navigateTo(currentItem.task, {})}}`
4. Click on button and see the issue

### Environment

Production

### Version

Cloud

Front logo [Front conversations](https://app.frontapp.com/open/top_56b2k)

### LINKED ISSUE #12833: [Bug]: `navigateTo` is throwing unexpected errors with list widget
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Screenshot 2022-04-12 at 12 11 03 PM
Screenshot 2022-04-12 at 12 43 53 PM

Here are the two instances of this issue. The `navigateTo` throws an error for valid bindings.

### Possible solution 
https://github.com/appsmithorg/appsmith/pull/14470

### Steps To Reproduce

https://theappsmith.slack.com/archives/C0341RERY4R/p1649716348012539

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by arunvjn**
## Description
The usage of single quote to parse the field values in a list widget opened up the possibility of XSS in list widget. 
The line  `{{((currentItem) => { ${next}})(JSON.parse('${escapedStringifiedListItem}'))}}` can be manipulated the following way. 
Let say, the escapedStringifiedListItem's value is a JSON that looks like this `{"username": '+showAlert(1.toString())+'}`, then  ``{{((currentItem) => { ${next}})(JSON.parse('{"username" :  '+showAlert(1.toString())+'}'))}}`` would parse incorrectly and showAlert would get executed when this snippet is sent for evaluation.

Fixes #9872 
Fixes #12833 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by jsartisan — DISMISSED** [review]

File: app/client/src/widgets/ListWidget/widget/index.tsx [hunk 1]
```diff
@@ -503,15 +501,12 @@ class ListWidget extends BaseWidget<ListWidgetProps<WidgetProps>, WidgetState> {
           const { jsSnippets } = getDynamicBindings(propertyValue);
           const listItem = this.props.listData?.[itemIndex] || {};
           const stringifiedListItem = JSON.stringify(listItem);
-          const escapedStringifiedListItem = escapeSpecialChars(
```
Comment (ashit-rath) [inline app/client/src/widgets/ListWidget/widget/index.tsx:506]: Why are we removing this?

Comment (ashit-rath) [reply app/client/src/widgets/ListWidget/widget/index.tsx:506]: Got it
So in that case can we rename the variable `escapedStringifiedListItem` to just `stringifiedListItem`?

Comment (ashit-rath) [reply app/client/src/widgets/ListWidget/widget/index.tsx:506]: Got it!

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/List/List7_spec.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  before(() => {
+    cy.addDsl(dsl);
+  });
+
+  it("1. Validate that list widget doesn't execute code", function() {
+    cy.PublishtheApp();
+    cy.wait(5000);
+    cy.get(".t--widget-inputwidgetv2 input")
+      .eq(1)
+      .type("'+(function() { while(true) {} })()+'", {
```
[the remark this answers is not in the document]
Comment (arunvjn) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/List/List7_spec.js]: Done

**Review by ashit-rath — APPROVED** [review]

**Pull request merged**